elixir/apps/domain/lib/domain/events/hooks/tokens.ex

```diff
@@ -3,11 +3,28 @@ defmodule Domain.Events.Hooks.Tokens do
     :ok
   end
 
-  def on_update(_old_data, _data) do
-    :ok
+  # updates for email tokens have no side effects
+  def on_update(%{"type" => "email"}, _data), do: :ok
+  def on_update(_old_data, %{"type" => "email"}), do: :ok
+
+  # Soft-delete
+  def on_update(%{"deleted_at" => nil} = old_data, %{"deleted_at" => deleted_at})
+      when not is_nil(deleted_at) do
+    on_delete(old_data)
   end
 
-  def on_delete(_old_data) do
+  # Regular update - not expected to happen in normal operation
+  def on_update(_old_data, _new_data) do
```

Comment:
If this isn't intended to happen, should we log a message if it does happen?

Reply:
I'm not sure, there's not really anything we care about if a Token changes? I guess we could log an error but would it really be considered an error?